turbinia/lib/libcloudforensics.py

```diff
@@ -143,17 +147,29 @@ def list_instances(self):
     Returns:
       Dictionary with name and metadata for each instance.
     """
-    operation = self.gce_api().instances().aggregatedList(
-        project=self.project_id).execute()
-    result = self.gce_operation(operation, zone=self.default_zone)
+    have_all = False
+    page_token = None
     instances = dict()
-    for zone in result['items']:
-      try:
-        for instance in result['items'][zone]['instances']:
-          _, zone = instance['zone'].rsplit('/', 1)
-          instances[instance['name']] = dict(zone=zone)
-      except KeyError:
-        pass
+    while not have_all:
```

Comment:
This code seems to be mostly repetetive in the different classes. Can we refactor to limit code duplication?

Fine to do in a follow up PR, but put in a TODO in that case.

Reply:
Yeah, I agree that there is duplicate code there.  I added a TODO for now -- I'm not sure what the best way to refactor that out would be.